### DISCUSSION TITLE
This page has expired due to inactivity. Would you like to refresh the page. (status code : 419)

### DISCUSSION THREAD

**Discussion by bhupindersingh007** [livewire/livewire#4435] [7. Bugs (archived)]
"laravel/framework": "^8.40",
"livewire/livewire": "^2.8"

i was trying to make a simple form with livewire like  [https://laravel-livewire.com/docs/2.x/input-validation](https://laravel-livewire.com/docs/2.x/input-validation), and then encounter following errors:

![1](https://user-images.githubusercontent.com/63149405/146931484-3a674b9a-c723-4fa1-913c-235a2b91d909.JPG)
![2](https://user-images.githubusercontent.com/63149405/146931492-165837f6-0dac-45e1-a03d-1c737e07826d.JPG)

I found out that mentioned errors occured because of csrf token that laravel is checking, as livewire events are sending post ajax request.

Then, i tried 

-  Meta Tag ` <meta name="csrf-token" value="{{ csrf_token() }}">` in page
-  `@csrf` blade directive in livewire component form: 

I was thinking maybe livewire intelligent about csrf (automatically take csrf in ajax request), but again same above mentioned errors occured. 

After that by using following solution, i was able to successfully store the data in database by form which is a livewire component
( no error occured).
[https://github.com/livewire/livewire/issues/831#issuecomment-627471725](https://github.com/livewire/livewire/issues/831#issuecomment-627471725)

But, i think disabling csrf token check / middleware for livewire components should not be considered as a solution as it may result in 
security issues. 

Is that issue with livewire or, i am missing something?

**Comment by joshhanley** [comment]
@bhupindersingh007 Livewire already adds your CSRF token to its requests for you, so you don't need to add it to your forms or anything.

Do you get the error straight away? Because it can also happen if the session times out.

Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?

Thanks!

**Reply by bhupindersingh007** [reply]
Hello,

Livewire example repo ( will reproduce the issue) : [https://github.com/bhupindersingh007/livewire-example](https://github.com/bhupindersingh007/livewire-example)

> Do you get the error straight away? 

Livewire shows alert message on input.

Here is example recording of mentioned issue:

![example](https://user-images.githubusercontent.com/63149405/147438357-82d5424c-755a-4d73-ab2c-471cee7fcdd0.gif)

Hope, This will help.
Thanks!
